avocado/utils/memory.py

```diff
@@ -27,8 +27,75 @@
 from . import genio
 
 
-# Returns total memory in kb
+def _check_memory_state(mem_block):
+    """
+    Check the given memory block is online or offline
+
+    :param mem_block: memory block id.
+    :type string: like 198
+    :return: 'True' if online or 'False' if offline
+    :rtype: bool
+    """
+    if 'online' in open('/sys/devices/system/memory/memory%s/state' % mem_block).read():
+        return True
+    return False
+
+
+def _check_hot_plug():
+    """
+    Check kernel support for memory hotplug
+
+    :return: True if hotplug supported,  else False
+    :rtype: 'bool'
+    """
+    if glob.glob('/sys/devices/system/memory/memory*'):
+        return True
+    return False
+
+
+def _is_hot_pluggable(mem_block):
+    """
+    Check if the given memory block is hotpluggable
+
+    :param mem_block: memory block id.
+    :type string: like 198
+    :retrun: True if hotpluggable, else False
+    :rtype: 'bool'
+    """
+    if os.path.exists('/sys/devices/system/memory/memory%s/removable' % mem_block):
```

Comment:
It's not the presence/absence of this file that represents the availability of the hot-plug feature. You have to check its content. 1 for hot-plug and 0 for non-hot-plug. If you know for a fact that in some systems this file can be missing for non-hot-plug blocks, than you have to test both the presence and the content of the file.